apps/files/src/components/DragAndDropNotice.vue

```diff
@@ -86,18 +88,29 @@ export default defineComponent({
 			}
 			return null
 		},
+
+		/**
+		 * Debounced function to reset the drag over state
+		 * Required as Firefox has a bug where no dragleave is emitted:
+		 * https://bugzilla.mozilla.org/show_bug.cgi?id=656164
+		 */
+		resetDragOver() {
+			return debounce(() => {
+				this.dragover = false
+			}, 3000)
+		},
 	},
 
 	mounted() {
 		// Add events on parent to cover both the table and DragAndDrop notice
-		const mainContent = window.document.querySelector('main.app-content') as HTMLElement
+		const mainContent = window.document.getElementById('app-content-vue') as HTMLElement
```

Comment:
To stay explicit about targeting a `main` element, as the var name suggest?

```suggestion
		const mainContent = window.document.getElementById('main#app-content-vue') as HTMLElement
```

Reply:
No as stated in the initial post the element tag name is not part of any contracts.
Only the ID is guaranteed to work, the idea behind is fixing this right now because with Vue3 this will break¹.
Side effect: Query by ID is much faster than including the tag name (does not matter that much but the query is factor 2-10 faster).

¹: Because our template looks like this `<main id="content">...</main>` and our Vue app will be mounted to `#content`.
With Vue2 the element would be replaced, so it looks like `<main id="app-content-vue">...</main>`.
But Vue3 will render within and not replace so it looks like `<main id="content"><div id="app-content-vue">...</div></main>`.